### PR TITLE
Update awslogs.md

### PR DESCRIPTION
**Pull request by mixja**
Signed-off-by: Justin Menga <justin.menga@gmail.com>

### Proposed changes

Add documentation for new AWS CloudWatch logs multiline pattern options

### Unreleased project version

Docker Engine 17.06

### Related issues

https://github.com/moby/moby/pull/30891

### CODE REVIEW

**Comment by mstanleyjones** [comment]
@nathanleclaire @FrenchBen @friism PTAL

File: engine/admin/logging/awslogs.md [hunk 1]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
 > **Note**:
 > Your AWS IAM policy must include the `logs:CreateLogGroup` permission before you attempt to use `awslogs-create-group`.
 
+### awslogs-multiline-pattern
+
+The `awslogs-multiline-pattern` specifies a regular expression pattern for identifying the start of a log message.  A log message is made of a line that matches the pattern and any following lines that don't match the pattern.
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: The `awslogs-multiline-pattern` specifies a regular expression pattern for identifying the beginning of a container log entry.  Any following lines that do not match the pattern are considered to be part of the current log entry. The next line that matches the pattern begins a new log entry.

File: engine/admin/logging/awslogs.md [hunk 2]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+### awslogs-multiline-pattern
+
+The `awslogs-multiline-pattern` specifies a regular expression pattern for identifying the start of a log message.  A log message is made of a line that matches the pattern and any following lines that don't match the pattern.  
+
+You cannot configure this option in conjunction with the `awslogs-datetime-format` option.
+
+> **Note**:
+> Multiline logging performs regular expression parsing and matching of all log messages, which may have a negative impact on logging performance.
+
+For example, to process the following log stream where new log messages start with the pattern `INFO`:
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: For example, to process the following log stream where new log entries begin with the pattern `INFO`:

File: engine/admin/logging/awslogs.md [hunk 3]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+
+```bash
+$ docker run --log-driver=awslogs \
+             --log-opt awslogs-region=us-east-1 \
+             --log-opt awslogs-group=myLogGroup \
+             --log-opt awslogs-multiline-pattern='^INFO' \
+             ...
+```
+
+Which will generate the following CloudWatch log events:
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: This configuration would generate the following CloudWatch log entries:

File: engine/admin/logging/awslogs.md [hunk 4]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+
+```
+# First event
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: Can you change "First event", "Second event", and "Third event" to use "entry" instead of event?

File: engine/admin/logging/awslogs.md [hunk 5]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+INFO Another multiline message was logged
+     Some random message
+
+# Third event
+INFO Another message was logged
+```
+
+### awslogs-datetime-format
+
+The `awslogs-datetime-format` specifies an [strftime expression pattern](http://strftime.org) for identifying the start of a log message.  A log message is made of a line that matches the pattern and any following lines that don't match the pattern.
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: The `awslogs-datetime-format` specifies a [strftime expression pattern](http://strftime.org) for for identifying the beginning of a container log entry. Any following lines that do not match the pattern are considered to be part of the current log entry. The next line that matches the pattern begins a new log entry.

File: engine/admin/logging/awslogs.md [hunk 6]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+INFO Another message was logged
+```
+
+### awslogs-datetime-format
+
+The `awslogs-datetime-format` specifies an [strftime expression pattern](http://strftime.org) for identifying the start of a log message.  A log message is made of a line that matches the pattern and any following lines that don't match the pattern.  
+
+You cannot configure this option in conjunction with the `awslogs-multiline-pattern` option.
+
+> **Note**:
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: This note is for multi-line. Should it be updated for this option, or was this a copy-paste artifact?

File: engine/admin/logging/awslogs.md [hunk 7]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+> Multiline logging performs regular expression parsing and matching of all log messages, which may have a negative impact on logging performance.
+
+For example, to process the following log stream where new log messages start with the following timestamps:
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: For example, to process the following log stream where new log entries begin with the following timestamps:

File: engine/admin/logging/awslogs.md [hunk 8]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+
+```
+[May 01, 2017 19:00:01] A message was logged
+[May 01, 2017 19:00:04] Another multiline message was logged
+Some random message
+with some random words
+[May 01, 2017 19:01:32] Another message was logged
+```
+
+An strftime expression of `[%b %d, %Y %H:%M:%S]` can be used:
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: You can use the `strftime` expression, `[%b %d, %Y %H:%M:%S]`:

File: engine/admin/logging/awslogs.md [hunk 9]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+
+```bash
+$ docker run --log-driver=awslogs \
+             --log-opt awslogs-region=us-east-1 \
+             --log-opt awslogs-group=myLogGroup \
+             --log-opt awslogs-datetime-format='[%b %d, %Y %H:%M:%S]' \
+             ...
+```
+
+Which will generate the following CloudWatch log events:
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: This configuration would generate the following CloudWatch log entries:

File: engine/admin/logging/awslogs.md [hunk 10]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+
+```
+# First event
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: Can you change "First event", "Second event", and "Third event" to use "entry" instead of event?

File: engine/admin/logging/awslogs.md [hunk 11]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
...
+# Second event
+[May 01, 2017 19:00:04] Another multiline message was logged
+Some random message
+with some random words
+
+# Third event
+[May 01, 2017 19:01:32] Another message was logged
+```
+
+The following strftime codes are supported:
```
Comment (nrdlngr) [inline engine/admin/logging/awslogs.md]: The following `strftime` codes are supported:

File: engine/admin/logging/awslogs.md (hunk 6 above)
Comment (mixja) [inline engine/admin/logging/awslogs.md]: This note is correct - you cannot reciprocally configure both options together

**Comment by mixja** [comment]
@nrdlngr - changes submitted as requested

**Comment by mstanleyjones** [comment]
@FrenchBen PTAL or nominate.

**Comment by FrenchBen** [comment]
I think this is more of a review by @samuelkarp - if he's happy with it, I don't see any blockers

File: docker-for-mac/networking.md [hunk 12]
```diff
@@ -78,11 +78,10 @@ There are two scenarios that the above limitations will affect:
 
 #### I want to connect from a container to a service on the host
 
-The Mac has a changing IP address (or none if you have no network access). Our
-current recommendation is to attach an unused IP to the `lo0` interface on the
-Mac; for example: `sudo ifconfig lo0 alias 10.200.10.1/24`, and make sure that
-your service is listening on this address or `0.0.0.0` (ie not `127.0.0.1`).
-Then containers can connect to this address.
+The Mac has a changing IP address (or none if you have no network access). From
```
Comment (samuelkarp) [inline docker-for-mac/networking.md]: This looks like an unrelated change.

File: engine/admin/logging/awslogs.md [hunk 13]
```diff
@@ -77,6 +77,115 @@ $ docker run --log-driver=awslogs \
 > **Note**:
 > Your AWS IAM policy must include the `logs:CreateLogGroup` permission before you attempt to use `awslogs-create-group`.
 
+### awslogs-multiline-pattern
+
```
Comment (samuelkarp) [inline engine/admin/logging/awslogs.md]: Does this need `{% include edge_only.md section"option" %}` or did the feature make it into the 17.06.0 RC?

Comment (mstanleyjones) [inline engine/admin/logging/awslogs.md]: It still needs it even if so, because 17.06 is not released yet. The code is:

```
{% include edge_only.md section="option" %}
```

Or maybe "feature" instead of "option" if that fits better.

**Comment by mstanleyjones** [comment]
I interactive rebased and removed the unrelated commit.

**Comment by mstanleyjones** [comment]
This was already addressed in #3615

**Comment by samuelkarp** [comment]
@mstanleyjones It looks like https://github.com/docker/docker.github.io/pull/3615 lost a lot of the content from this one.  Is that intentional?

**Comment by mstanleyjones** [comment]
Not necessarily, no. I'll review it and see what we need to keep from here. I wasn't focused on this PR when I did the work on that other one.

**Comment by mstanleyjones** [comment]
I'm taking this over to get the important info in here ported over.

File: engine/admin/logging/awslogs.md [hunk 14]
```diff
@@ -97,39 +97,132 @@ $ docker run --log-driver=awslogs \
...
+patther `INFO`:
+
+```none
+INFO A message was logged
+INFO Another multiline message was logged
+     Some random message
+INFO Another message was logged
+```
+
+You can use the strftime expression of `[%b %d, %Y %H:%M:%S]`:
```
Comment (samuelkarp) [inline engine/admin/logging/awslogs.md]: Looks like this line is misplaced.  Did you mean to have it say "You can use the regular expression of `^INFO`:"?

File: engine/admin/logging/awslogs.md [hunk 15]
```diff
@@ -97,39 +97,132 @@ $ docker run --log-driver=awslogs \
...
+This option is ignored if `awslogs-datetime-format` is also configured.
+
+> **Note**:
+> Multiline logging performs regular expression parsing and matching of all log
+> messages. This may have a negative impact on logging performance.
+
+For example, to process the following log stream where new log messages start with the pattern `INFO`:
+
+Consider the following log stream, where each log message should start with the
+patther `INFO`:
```
Comment (samuelkarp) [inline engine/admin/logging/awslogs.md]: s/patther/pattern/
